integrationTests/multiShard/endOfEpoch/executingEpochChange_test.go

```diff
@@ -131,3 +132,138 @@ func TestEndOfEpochChangeWithoutTransactionInMultiShardedEnvironment(t *testing.
 		}
 	}
 }
+
+func TestEndOfEpochChangeWithContinuousTransactionsInMultiShardedEnvironment(t *testing.T) {
+	if testing.Short() {
+		t.Skip("this is not a short test")
+	}
+
+	numOfShards := 2
+	nodesPerShard := 3
+	numMetachainNodes := 3
+
+	advertiser := integrationTests.CreateMessengerWithKadDht(context.Background(), "")
+	_ = advertiser.Bootstrap()
+
+	logger.DefaultLogger().SetLevel("DEBUG")
+
+	nodes := integrationTests.CreateNodes(
+		numOfShards,
+		nodesPerShard,
+		numMetachainNodes,
+		integrationTests.GetConnectableAddress(advertiser),
+	)
+
+	roundsPerEpoch := int64(20)
+	for _, node := range nodes {
+		node.EndOfEpochTrigger.SetRoundsPerEpoch(roundsPerEpoch)
+	}
+
+	idxProposers := make([]int, numOfShards+1)
+	for i := 0; i < numOfShards; i++ {
+		idxProposers[i] = i * nodesPerShard
+	}
+	idxProposers[numOfShards] = numOfShards * nodesPerShard
+
+	integrationTests.DisplayAndStartNodes(nodes)
+
+	defer func() {
+		_ = advertiser.Close()
+		for _, n := range nodes {
+			_ = n.Node.Stop()
+		}
+	}()
+
+	initialVal := big.NewInt(10000000)
+	sendValue := big.NewInt(5)
+	integrationTests.MintAllNodes(nodes, initialVal)
+	receiverAddress := []byte("12345678901234567890123456789012")
+
+	round := uint64(0)
+	nonce := uint64(0)
+	round = integrationTests.IncrementAndPrintRound(round)
+	nonce++
+
+	time.Sleep(time.Second)
+
+	/////////----- wait for epoch end period
+	for i := uint64(0); i <= uint64(roundsPerEpoch); i++ {
+		integrationTests.ProposeBlock(nodes, idxProposers, round, nonce)
+		integrationTests.SyncBlock(t, nodes, idxProposers, round)
+		round = integrationTests.IncrementAndPrintRound(round)
+		nonce++
+
+		for _, node := range nodes {
+			integrationTests.CreateAndSendTransaction(node, sendValue, receiverAddress, "")
+		}
+
+		time.Sleep(time.Second)
+	}
+
+	time.Sleep(time.Second)
+
+	nrRoundsToPropagateMultiShard := 10
+	for i := 0; i < nrRoundsToPropagateMultiShard; i++ {
```

Comment:
this block of code is similar to previous, you can extract this in a function

Reply:
done